Code actions don't work

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

<img width="1219" alt="image" src="https://github.com/user-attachments/assets/395e315d-9225-46a6-98be-42ae8e2468d5">

When I click the run button next to "fn main" it starts terminal with this output:

```sh
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.13s
     Running `/Users/artem/dev/rust/l2Rust/target/debug/game`
thread 'main' panicked at src/game_server/dto/config.rs:21:33:
Failed to open config file: config/game.yaml. Error: No such file or directory (os error 2)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

⏵ Task `cargo run -p L2Rust --bin game` finished with non-zero error code: 101
⏵ Command: /opt/homebrew/bin/fish -i -c 'cargo run -p L2Rust --bin game'
```

The fix is to add to the path in the code `../config/game.yaml` but it breaks the build and `cargo build` doesn't work (and it's expected as it's looking for config dir in root project directory which is above src folder.)
Here is a repo I am working on: https://github.com/artemijan/L2Rust


### Environment

Zed: v0.161.2 (Zed)
OS: macOS 15.1.0
Memory: 64 GiB
Architecture: aarch64
Shell: fish

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary>

<!-- Click below this line and paste or drag-and-drop your log-->
```
2024-11-19T20:16:35.697713+01:00 [INFO] ========== starting zed ==========
2024-11-19T20:16:35.699484+01:00 [INFO] Opening main db
2024-11-19T20:16:35.700348+01:00 [INFO] Opening main db
2024-11-19T20:16:35.722383+01:00 [INFO] Using git binary path: Some("/Applications/Zed.app/Contents/MacOS/git")
2024-11-19T20:16:35.892129+01:00 [INFO] no tokio runtime found, creating one for Reqwest...
2024-11-19T20:16:35.909978+01:00 [ERROR] theme not found: VSCode Dark Modern
2024-11-19T20:16:35.911166+01:00 [INFO] initializing context server client
2024-11-19T20:16:35.911422+01:00 [INFO] Registering built-in prompt template: content_prompt
2024-11-19T20:16:35.911798+01:00 [INFO] Registering built-in prompt template: project_slash_command
2024-11-19T20:16:35.911854+01:00 [INFO] Registering built-in prompt template: suggest_edits
2024-11-19T20:16:35.911952+01:00 [INFO] Registering built-in prompt template: terminal_assistant_prompt
2024-11-19T20:16:35.912086+01:00 [INFO] No prompt template overrides directory found at /Users/artem/.config/zed/prompt_overrides. Using built-in prompts.
2024-11-19T20:16:35.913394+01:00 [INFO] extensions updated. loading 3, reloading 0, unloading 0
2024-11-19T20:16:35.937075+01:00 [ERROR] invalid binding value for keystroke alt-shift-up, context Some("Editor")

Caused by:
    no action type registered for editor::DuplicateLine
2024-11-19T20:16:35.937347+01:00 [ERROR] invalid binding value for keystroke escape, context Some("ChatPanel > MessageEditor")

Caused by:
    no action type registered for chat_panel::CloseReplyPreview
2024-11-19T20:16:35.938966+01:00 [INFO] Opening main db
2024-11-19T20:16:35.942654+01:00 [WARN] Theme "Material Dark" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-11-19T20:16:35.942871+01:00 [ERROR] theme not found: VSCode Dark Modern
2024-11-19T20:16:35.944049+01:00 [INFO] building git repository, `.git` path in the worktree: ".git"
2024-11-19T20:16:35.944211+01:00 [ERROR] theme not found: VSCode Dark Modern
2024-11-19T20:16:35.977196+01:00 [INFO] Opening main db
2024-11-19T20:16:35.979269+01:00 [INFO] set status on client 0: Authenticating
2024-11-19T20:16:35.979965+01:00 [INFO] open paths ["/Users/artem/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/fmt/mod.rs"]
2024-11-19T20:16:35.980019+01:00 [INFO] open paths ["/Users/artem/.config/zed/settings.json"]
2024-11-19T20:16:35.980065+01:00 [INFO] open paths ["/Users/artem/.config/zed/tasks.json"]
2024-11-19T20:16:35.980514+01:00 [INFO] Opening main db
2024-11-19T20:16:35.992997+01:00 [INFO] set status on client 112002: Connecting
2024-11-19T20:16:36.000678+01:00 [INFO] attempting to start language server "json-language-server", path: "/Users/artem/.config/zed/settings.json", id: 1
2024-11-19T20:16:36.000772+01:00 [INFO] Initializing default prettier with plugins {}
2024-11-19T20:16:36.000879+01:00 [INFO] Initializing default prettier with plugins {}
2024-11-19T20:16:36.001276+01:00 [INFO] fetching latest version of language server "json-language-server"
2024-11-19T20:16:36.001358+01:00 [INFO] Node runtime install_if_needed
2024-11-19T20:16:36.009307+01:00 [INFO] attempting to start language server "json-language-server", path: "/Users/artem/.config/zed/tasks.json", id: 2
2024-11-19T20:16:36.009389+01:00 [INFO] Initializing default prettier with plugins {}
2024-11-19T20:16:36.030037+01:00 [INFO] attempting to start language server "rust-analyzer", path: "/Users/artem/dev/rust/l2Rust", id: 3
2024-11-19T20:16:36.031479+01:00 [INFO] attempting to start language server "rust-analyzer", path: "/Users/artem/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/fmt/mod.rs", id: 4
2024-11-19T20:16:36.031635+01:00 [INFO] Initializing default prettier with plugins {}
2024-11-19T20:16:36.031684+01:00 [INFO] Initializing default prettier with plugins {}
2024-11-19T20:16:36.072058+01:00 [INFO] Initializing default prettier with plugins {}
2024-11-19T20:16:36.07218+01:00 [INFO] Initializing default prettier with plugins {}
2024-11-19T20:16:36.130195+01:00 [INFO] set environment variables from shell:/opt/homebrew/bin/fish, path:/Users/artem/.local/bin:/Users/artem/Library/Application Support/JetBrains/Toolbox/scripts:/opt/homebrew/opt/llvm/bin:/Users/artem/.modular/pkg/packages.modular.com_mojo/bin:/Users/artem/.pyenv/shims:/Users/artem/.sdkman/candidates/maven/current/bin:/Users/artem/.sdkman/candidates/java/current/bin:/Users/artem/.modular/bin:/Users/artem/.cargo/bin:/opt/homebrew/bin:/opt/homebrew/opt/fzf/bin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin
2024-11-19T20:16:36.19666+01:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-11-19T20:16:36.317579+01:00 [INFO] using project environment variables shell launched in "/Users/artem/.config/zed/settings.json". PATH="/Users/artem/.local/bin:/Users/artem/Library/Application Support/JetBrains/Toolbox/scripts:/opt/homebrew/opt/llvm/bin:/Users/artem/.modular/pkg/packages.modular.com_mojo/bin:/Users/artem/.pyenv/shims:/Users/artem/.sdkman/candidates/maven/current/bin:/Users/artem/.sdkman/candidates/java/current/bin:/Users/artem/.modular/bin:/Users/artem/.cargo/bin:/opt/homebrew/bin:/opt/homebrew/opt/fzf/bin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin"
2024-11-19T20:16:36.317718+01:00 [INFO] found rust-analyzer in PATH. trying to run `rust-analyzer --help`
2024-11-19T20:16:36.53811+01:00 [INFO] starting language server process. binary path: "/opt/homebrew/bin/node", working directory: "/", args: ["/Users/artem/Library/Application Support/Zed/copilot/copilot-v0.5.0/dist/agent.js", "--stdio"]
2024-11-19T20:16:36.619085+01:00 [INFO] downloading language server "json-language-server"
2024-11-19T20:16:36.732609+01:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-11-19T19:16:36.730Z] Agent service starting",
  "metadataStr": "[DEBUG] [agent] [2024-11-19T19:16:36.730Z]",
  "extra": [
    "Agent service starting"
  ]
}
2024-11-19T20:16:36.733432+01:00 [INFO] Language server with id 0 sent unhandled notification client/registerCapability:
{
  "registrations": [
    {
      "id": "a745a4c0-b362-475a-8d95-757a03efe9d9",
      "method": "workspace/didChangeWorkspaceFolders",
      "registerOptions": {}
    }
  ]
}
2024-11-19T20:16:36.736534+01:00 [INFO] add connection to peer
2024-11-19T20:16:36.736567+01:00 [INFO] add_connection;
2024-11-19T20:16:36.736594+01:00 [INFO] waiting for server hello
2024-11-19T20:16:36.746678+01:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-11-19T19:16:36.736Z] Telemetry initialized",
  "metadataStr": "[DEBUG] [agent] [2024-11-19T19:16:36.736Z]",
  "extra": [
    "Telemetry initialized"
  ]
}
2024-11-19T20:16:36.760088+01:00 [INFO] got server hello
2024-11-19T20:16:36.760132+01:00 [INFO] set status to connected (connection id: ConnectionId { owner_id: 0, id: 0 }, peer id: PeerId { owner_id: 611, id: 692911 })
2024-11-19T20:16:36.76018+01:00 [INFO] set status on client 112002: Connected { peer_id: PeerId { owner_id: 611, id: 692911 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }
2024-11-19T20:16:36.791974+01:00 [ERROR] failed to run rust-analyzer after detecting it in PATH: binary: "/Users/artem/.cargo/bin/rust-analyzer": exit status: 1, stdout: "", stderr: "error: Unknown binary 'rust-analyzer' in official toolchain 'stable-aarch64-apple-darwin'.\n"
2024-11-19T20:16:36.792041+01:00 [INFO] fetching latest version of language server "rust-analyzer"
2024-11-19T20:16:36.82744+01:00 [INFO] using project environment for language server LanguageServerName("json-language-server")
2024-11-19T20:16:36.827507+01:00 [INFO] using project environment for language server LanguageServerName("json-language-server")
2024-11-19T20:16:36.827561+01:00 [INFO] starting language server process. binary path: "/opt/homebrew/bin/node", working directory: "/Users/artem/.config/zed", args: ["/Users/artem/Library/Application Support/Zed/languages/json-language-server/node_modules/vscode-langservers-extracted/bin/vscode-json-language-server", "--stdio"]
2024-11-19T20:16:36.827836+01:00 [INFO] starting language server process. binary path: "/opt/homebrew/bin/node", working directory: "/Users/artem/.config/zed", args: ["/Users/artem/Library/Application Support/Zed/languages/json-language-server/node_modules/vscode-langservers-extracted/bin/vscode-json-language-server", "--stdio"]
2024-11-19T20:16:37.736114+01:00 [INFO] downloading language server "rust-analyzer"
2024-11-19T20:16:37.740707+01:00 [INFO] using project environment for language server LanguageServerName("rust-analyzer")
2024-11-19T20:16:37.741023+01:00 [INFO] found rust-analyzer in PATH. trying to run `rust-analyzer --help`
2024-11-19T20:16:37.741426+01:00 [ERROR] failed to run rust-analyzer after detecting it in PATH: binary: "/Users/artem/.cargo/bin/rust-analyzer": No such file or directory (os error 2)
2024-11-19T20:16:37.74157+01:00 [INFO] using project environment for language server LanguageServerName("rust-analyzer")
2024-11-19T20:16:37.741686+01:00 [INFO] starting language server process. binary path: "/Users/artem/Library/Application Support/Zed/languages/rust-analyzer/rust-analyzer-2024-11-18", working directory: "/Users/artem/dev/rust/l2Rust", args: []
2024-11-19T20:16:37.743141+01:00 [INFO] starting language server process. binary path: "/Users/artem/Library/Application Support/Zed/languages/rust-analyzer/rust-analyzer-2024-11-18", working directory: "/Users/artem/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/fmt", args: []
2024-11-19T20:16:38.332317+01:00 [ERROR] Language server rust-analyzer-2024-11-18 (id 4) status update: Failed to load workspaces.
2024-11-19T20:16:39.124079+01:00 [ERROR] Language server rust-analyzer-2024-11-18 (id 4) status update: Failed to load workspaces.
2024-11-19T20:16:45.926852+01:00 [INFO] open paths ["/Users/artem/.config/zed/settings.json"]
2024-11-19T20:17:03.460021+01:00 [INFO] Initializing default prettier with plugins {}
2024-11-19T20:17:03.460226+01:00 [INFO] Initializing default prettier with plugins {}

```
<!-- Click above this line and paste or drag-and-drop your log--></details>

--- a/crates/languages/src/rust.rs
+++ b/crates/languages/src/rust.rs
@@ -882,7 +882,7 @@ impl ContextProvider for RustContextProvider {
                     RUST_BIN_REQUIRED_FEATURES_FLAG_TASK_VARIABLE.template_value(),
                     RUST_BIN_REQUIRED_FEATURES_TASK_VARIABLE.template_value(),
                 ],
-                cwd: Some("$ZED_DIRNAME".to_owned()),
+                cwd: Some(RUST_MANIFEST_DIRNAME_TASK_VARIABLE.template_value()),
                 tags: vec!["rust-main".to_owned()],
                 ..TaskTemplate::default()
             },
@@ -904,14 +904,14 @@ impl ContextProvider for RustContextProvider {
                 label: "Run".into(),
                 command: "cargo".into(),
                 args: run_task_args,
-                cwd: Some("$ZED_DIRNAME".to_owned()),
+                cwd: Some(RUST_MANIFEST_DIRNAME_TASK_VARIABLE.template_value()),
                 ..TaskTemplate::default()
             },
             TaskTemplate {
                 label: "Clean".into(),
                 command: "cargo".into(),
                 args: vec!["clean".into()],
-                cwd: Some("$ZED_DIRNAME".to_owned()),
+                cwd: Some(RUST_MANIFEST_DIRNAME_TASK_VARIABLE.template_value()),
                 ..TaskTemplate::default()
             },
         ];

